workflows/const.go

```diff
@@ -127,3 +164,25 @@ func simplifyRepoURL(url string) string {
 
 	return url[slashIndex+1:]
 }
+
+func concatTagMaps(ymlMap map[string]interface{}, muMap map[string]string, constMap map[string]string) (map[string]string, error) {
```

Comment:
Change this to take 2 params: interface and a map.  The interface is one of the structs declared above.  The map is the custom key/values for the tags.

Then, use reflection to add all the values from the struct.  Merge in any values from the map, but error if trying to use a "reserved" key.